osquery/tables/system/darwin/certificates.mm

```diff
@@ -25,39 +25,108 @@
 void genCertificate(X509* cert, const std::string& path, QueryData& results) {
   Row r;
 
-  // Generate the common name and subject.
-  // They are very similar OpenSSL API accessors so save some logic and
-  // generate them using output parameters.
-  genCommonName(cert, r["subject"], r["common_name"], r["issuer"]);
-  // Same with algorithm strings.
-  genAlgorithmProperties(
-      cert, r["key_algorithm"], r["signing_algorithm"], r["key_strength"]);
-
-  // Most certificate field accessors return strings.
-  r["not_valid_before"] = INTEGER(genEpoch(X509_get_notBefore(cert)));
-  r["not_valid_after"] = INTEGER(genEpoch(X509_get_notAfter(cert)));
+  auto opt_issuer_name = getCertificateIssuerName(cert);
+  if (opt_issuer_name.has_value()) {
+    r["issuer"] = SQL_TEXT(opt_issuer_name.value());
+  } else {
+    r["issuer"] = SQL_TEXT("");
+  }
```

Comment:
Do we need the `else` condition for any of these? Feels more correct to let it return the NULL there

Reply:
The instances where this is happening across the codebase are slowly being fixed; the problem with leaving unpopulated columns is that what happens next depends on where the query is being executed:

1. Internal SQL APIs (such as `selectAllFrom`): Those columns will remain unpopulated and accessing them will crash osquery
2. Everything else: a default value (like a 0, or an empty string) will be silently inserted

There have been problems in the past where columns were sometimes being skipped by a platform but not the others, causing crashes that were hard to debug.